### PR TITLE
Textual signing

### PR DESCRIPTION
**Pull request by joneskm**

### CODE REVIEW

File: x/auth/src/ante/ante.rs [hunk 1]
```diff
@@ -275,21 +284,60 @@ impl<BK: BankKeeper<SK>, AK: AuthKeeper<SK>, SK: StoreKey> BaseAnteHandler<BK, A
...
+                        };
+
+                        let _tx_data = TxData {
+                            body: tx.tx.body.clone(),
+                            auth_info: tx.tx.auth_info.clone(),
+                        };
+
+                        //handler.sign_bytes_get(ctx, signer_data, tx_data).unwrap()
+                        //TODO: remove unwrap
+                        todo!()
```
Comment (NYBACHOK) [inline x/auth/src/ante/ante.rs:324]: Is implementing this code should be part of this PR?

File: x/auth/src/signing/renderer/messages/msg_send.rs [hunk 2]
```diff
@@ -83,38 +79,40 @@ mod tests {
...
+    use crate::signing::renderer::value_renderer::ValueRenderer;
 
-    #[test]
-    fn screen_result_no_coins() -> anyhow::Result<()> {
-        const MESSAGE: &str = r#"{
-            "from_address": "cosmos1ulav3hsenupswqfkw2y3sup5kgtqwnvqa8eyhs",
-            "to_address": "cosmos1ejrf4cur2wy6kfurg9f2jppp2h3afe5h6pkh5t",
-            "amount": []
-        }"#;
+    // TODO: fix this test
```
Comment (NYBACHOK) [inline x/auth/src/signing/renderer/messages/msg_send.rs:84]: Should I check this one?

File: x/bank/src/abci_handler.rs [hunk 3]
```diff
@@ -55,7 +56,17 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey> ABCIHandler<SK, PSK> {
 
                 Ok(self.keeper.query_balance(ctx, req).encode_vec().into())
             }
-
+            "/cosmos.bank.v1beta1.Query/DenomsMetadata" => {
```
Comment (NYBACHOK) [inline x/bank/src/abci_handler.rs:59]: I think you should make const with this str's

File: x/bank/src/client/cli/query.rs [hunk 4]
```diff
@@ -43,6 +48,18 @@ pub fn run_bank_query_command(
...
             )?;
 
+            Ok(serde_json::to_string_pretty(&res)?)
+        }
+        BankCommands::DenomMetadata => {
+            let query = QueryDenomsMetadataRequest { pagination: None };
+
+            let res = run_query::<QueryDenomsMetadataResponse, RawQueryDenomsMetadataResponse>(
+                query.encode_to_vec(),
+                "/cosmos.bank.v1beta1.Query/DenomsMetadata".into(),
```
Comment (NYBACHOK) [inline x/bank/src/client/cli/query.rs:58]: Here too

File: x/auth/src/signing/handler.rs [hunk 5]
```diff
@@ -1,27 +1,25 @@
...
-    renderer::value_renderer::{DefaultValueRenderer, ValueRenderer},
-};
+use super::{errors::SigningErrors, renderer::value_renderer::ValueRenderer};
 
 #[derive(Debug)]
 pub struct SignModeHandler;
 
 impl SignModeHandler {
-    pub fn sign_bytes_get<SK: StoreKey>(
+    pub fn sign_bytes_get<SK: StoreKey, DB: Database>(
```
Comment (NYBACHOK) [inline x/auth/src/signing/handler.rs:18]: So you have chosen to remove possibility of different modes?

**Review by NYBACHOK — APPROVED** [review]

File: x/auth/src/ante/ante.rs (hunk 1 above)
Comment (joneskm) [reply x/auth/src/ante/ante.rs:324]: This PR is getting a bit big. So I'll do that as a separate PR.

File: x/auth/src/signing/renderer/messages/msg_send.rs (hunk 2 above)
Comment (joneskm) [reply x/auth/src/signing/renderer/messages/msg_send.rs:84]: You have a PR which fixes this.

File: x/bank/src/client/cli/query.rs (hunk 4 above)
Comment (joneskm) [reply x/bank/src/client/cli/query.rs:58]: Yes, as above.

**Pull request merged**